demo_nodes_cpp_native/CMakeLists.txt

```diff
@@ -11,25 +11,32 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
 
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
 find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)
+find_package(rclcpp_components REQUIRED)
 find_package(rmw REQUIRED)
 find_package(std_msgs REQUIRED)
 find_package(rmw_fastrtps_cpp QUIET)
 
-function(custom_executable target)
-  add_executable(${target} src/${target}.cpp)
-  ament_target_dependencies(${target}
-    "rclcpp"
-    "std_msgs"
-    "rmw_fastrtps_cpp")
-  install(TARGETS ${target}
-    DESTINATION lib/${PROJECT_NAME})
-endfunction()
+add_library(talker_native SHARED
+  src/talker.cpp)
+ament_target_dependencies(talker_native
+  "rclcpp"
+  "std_msgs"
+  "rclcpp_components"
+  "rmw_fastrtps_cpp")
```

Comment:
This change fails the build when FastRTPS is not available:

```
CMake Error at /home/jenkins-agent/workspace/ci_linux/ws/install/ament_cmake_target_dependencies/share/ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake:52 (message):
  ament_target_dependencies() the passed package name 'rmw_fastrtps_cpp' was
  not found before
```

Please fix asap.

Reply:
Resolved in https://github.com/ros2/demos/pull/381

Reply:
@skucheria I think it's best to fix this is a PR separate from #381 so we can get it in sooner.